File: GGAndroid_2.0/GGCore/GGLib/src/main/java/io/ganguo/library/ui/view/ActivityInterface.java
package io.ganguo.library.ui.view;

import android.app.Activity;
import android.databinding.ViewDataBinding;
import android.os.Bundle;

/**
 * Created by Roger on 6/7/16.
 */
public interface ActivityInterface<T extends ViewDataBinding> extends ViewInterface<T> {
    Activity getActivity();

    Bundle getBundle();
}

File: GGAndroid_4.0/core/frame/frame-mvvm/src/main/kotlin/io/ganguo/mvvm/viewinterface/ActivityInterface.kt
package io.ganguo.mvvm.viewinterface

import android.content.Intent
import android.os.Bundle
import androidx.activity.result.ActivityResult
import androidx.databinding.ViewDataBinding
import androidx.fragment.app.FragmentActivity
import io.ganguo.rxresult.GGActivityResultLauncher

/**
 * <pre>
 *     author : leo
 *     time   : 2019/10/11
 *     desc   : Activity Interface
 * </pre>
 */
interface ActivityInterface<T : ViewDataBinding> : ViewInterface<T> {
    val activity: FragmentActivity
    val resultLauncher: GGActivityResultLauncher<Intent, ActivityResult>
    val bundle: Bundle?
}